File: Go/0006.go
func convert(s string, numRows int) string {
	if numRows == 1 || numRows >= len(s) {
		return s
	}

	rows := make([][]rune, numRows)
	direction := 1 // 1 for down, -1 for up
	row := 0

	for _, char := range s {
		rows[row] = append(rows[row], char)

		if row == 0 {
			direction = 1
		} else if row == numRows-1 {
			direction = -1
		}

		row += direction
	}

	var result []rune
	for _, r := range rows {
		result = append(result, r...)
	}

	return string(result)
}

File: Java/0006.java
class Solution {
    public String convert(String s, int numRows) {
        if (numRows == 1 || numRows >= s.length()) {
            return s;
        }

        StringBuilder[] rows = new StringBuilder[numRows];
        for (int i = 0; i < numRows; i++) {
            rows[i] = new StringBuilder();
        }

        int direction = 1; // 1 for down, -1 for up
        int row = 0;

        for (char c : s.toCharArray()) {
            rows[row].append(c);

            if (row == 0) {
                direction = 1;
            } else if (row == numRows - 1) {
                direction = -1;
            }

            row += direction;
        }

        StringBuilder result = new StringBuilder();
        for (StringBuilder sb : rows) {
            result.append(sb);
        }

        return result.toString();
    }
}
